system/threads.cpp

```diff
@@ -115,44 +166,155 @@ long __syscall_futex(uint32_t* uaddr, int futex_op, ...)
 				if (*uaddr != val)
 				{
 					futexSpinLock.unlock();
-					return EAGAIN;
+					return -EAGAIN;
```

Comment:
let's put these changes in a separate commit before the main one

Reply:
Done